src/LinuxTracing/SwitchesStatesNamesVisitor.cpp

```diff
@@ -173,14 +175,27 @@ void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
 }
 
 void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
-                                       const SchedWakeupPerfEventData& event_data) {
+                                       const SchedSwitchPerfEventData& event_data) {
+  VisitSchedSwitch(event_timestamp, event_data, /*wait_for_callstack=*/false);
+}
+
+void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
+                                       const SchedSwitchWithStackPerfEventData& event_data) {
+  const bool wait_for_callstack = event_data.data != nullptr;
```

Comment:
Why would this be `nullptr` if we have `SchedSwitchWithStackPerfEventData`? Why would we not be using `SchedSwitchPerfEventData` in that case?

Reply:
There are several cases, where we had requested to get callstacks from perf_event_open, but may not receive one.

E.g. for events triggered by "pid 0", we often get no callstack. Also in the near future, we do not want to copy the stack on events that were on different pids.

So you would prefer to adjust the consumer, to not produce SchedSwitchWithStackPerfEvents at that point?

Reply:
Then I think this is basically what I wrote here: https://github.com/google/orbit/pull/4118#discussion_r955847533

In the cases you mention we should just produce a `SchedSwitchPerfEventData`. Having both `SchedSwitchWithStackPerfEventData` and `SchedSwitchPerfEventData` is otherwise useless. I'm assuming this is possible but I don't know the details, i.e., when exactly we know that a stack is included or not.

Reply:
Should be totally, doable. In the consumer, we get the perf "mega" struct that contains all our fields. We check if we have a callstack, and if not, we create a SchedSwitchPerfEvent instead.

Reply:
Changed the "reader" to produce non-callstack events in the case the callstack or the registers are missing (i.e. that unwinding would fail).